core/trino-main/src/test/java/io/trino/operator/output/TestPartitionedOutputOperator.java

```diff
@@ -357,6 +357,37 @@ public void testOutputForSimplePageWithType(Type type)
         assertThat(partitioned).containsExactlyInAnyOrderElementsOf(expected); // order is different due to 2 partitions joined
     }
 
+    @Test(dataProvider = "types")
+    public void testOutputForMixedSmallAndNormalPagesWithType(Type type)
```

Comment:
we have special support for small pages using the old row-wise approach. This tests that interleaving small and normal pages works together correctly.

Reply:
Comment is needed.

Can you UT both approaches explicitly, e.g. via exposing them with `@VisibleForTesting`?

Reply:
Comment added.
Re UT.  Why would we need to test it explicitly?

Reply:
I don't think it's a good practice to expose private members to be sure they are tested if they can be easily tested using public api.

Reply:
> if they can be easily tested using public api.

The thing is you don't know if they will be tested in 6 months, because condition for going particular path (e.g. `COLUMNAR_STRATEGY_COEFFICIENT` could change). So your test is fragile unless you can validate particular path is executed. You might loose coverage and not even know about it.

> I don't think it's a good practice to expose private members to be sure they are tested

You don't make them public. You make them package private and annotate with `@VisibleForTesting`. We use that approach in many places.

Reply:
> The thing is you don't know if they will be tested in 6 months, because condition for going particular path (e.g. COLUMNAR_STRATEGY_COEFFICIENT could change). So your test is fragile unless you can validate particular path is executed. You might loose coverage and not even know about it.

This is true for any part of the code, yet we do not explicitly test all the private methods, or for example insides of if statements, and rightly so.